coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -5439,21 +5439,32 @@ def baddbmm(context, node):
     inputs = _get_inputs(context, node, expected=5)
     bias, batch1, batch2, beta, alpha = inputs
 
-    if beta.val != 1.0:
-        # Apply scaling factor beta to the bias.
-        bias = mb.mul(x=beta, y=bias, name=bias.name + "_scaled")
-        context.add(bias)
-
     if alpha.val != 1.0:
         # Apply scaling factor alpha to the input.
         batch1 = mb.mul(x=alpha, y=batch1, name=batch1.name + "_scaled")
         context.add(batch1)
 
     bmm_node = mb.matmul(x=batch1, y=batch2, name=node.name + "_bmm")
-    context.add(bmm_node)
 
-    baddbmm_node = mb.add(x=bias, y=bmm_node, name=node.name)
-    context.add(baddbmm_node)
+    if beta.val != 0.0 or bias.shape != bmm_node.shape:
+        context.add(bmm_node)
+        if beta.val != 1.0:
+            # Torch supports integers, so convert to float before
+            if beta.dtype != bias.dtype:
+                logger.warning(
+                    f"Casting the `beta`(value={beta.val}) argument of `baddbmm` op {node.name} "
+                    f"from {beta.dtype} to {bias.dtype} dtype")
+                beta = mb.cast(x=beta, dtype=TYPE_TO_DTYPE_STRING[bias.dtype])
+            # Apply scaling factor beta to the bias.
+            bias = mb.mul(x=beta, y=bias, name=bias.name + "_scaled")
+            context.add(bias)
+
+        baddbmm_node = mb.add(x=bias, y=bmm_node, name=node.name)
+        context.add(baddbmm_node)    
+    else:
+        bmm_node.name = node.name
+        context.add(bmm_node)
+
```

Comment:
This change looks good to me, but I'm not sure why we would add `bmm_node` to context in addition to `baddbmm_node`. If @TobyRoseman can confirm that there is no hidden reason that I'm unaware of, I would be happy to approve this PR

Reply:
I don't know either. Let's just make sure this change is well tested, and I think it is well tested..

Reply:
So you mean that in general `context.add(baddbmm_node)` should be enough in the bias branch, without having to do  `context.add(bmm_node)` ? I've just preserved what used to be there before and indeed it seems to work locally when running the CI tests. Does it mean that the `context.add(bias)` is also not necessary as well and that in general there should be a single `context.add(node_with_input_node_name)` per op ?